apis/common/v1/resource.go

```diff
@@ -77,10 +77,48 @@ type SecretKeySelector struct {
 	Key string `json:"key"`
 }
 
+// Policy represents the Resolve and Resolution policies of Reference instance.
+type Policy struct {
+	// Resolve specifies when this reference should be resolved. The default
+	// is 'IfNotPresent', which will attempt to resolve the reference only when
+	// the corresponding field is not present. Use 'Always' to resolve the
+	// reference on every reconcile.
+	// +optional
+	Resolve *ResolvePolicy `json:"resolve,omitempty"`
```

Comment:
We can make this enum.

Reply:
Would it make sense for it to be possible to explicitly specify `IfNotPresent`? Currently that's not possible because the enum validation only allows `Always`.

Reply:
Yes, I think enum should include all possibilities. @sergenyalcin would you mind opening a PR to add this?